tests/test_interpolate.py

```diff
@@ -0,0 +1,227 @@
+    Periodic torsions differ from bonds and angles in that their uniqueness is
+    also determined by the "period", which is currently encoded as one of the parameters.
+
+    We expect that decoupled terms have their force constants turned set to zero,
+    while maintaining the same equilibrium angles for the *same* period.
+    """
+    a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p = np.random.rand(16)
+
+    # tbd: what do we do if there are repeats?
+    # merge repeats into a single term first?
+    src_idxs = [(2, 3, 9, 4), (2, 1, 4, 3), (0, 1, 4, 2), (0, 1, 4, 2)]
+    src_params = [(a, b, 2), (c, d, 1), (e, f, 3), (g, h, 1)]
+
+    dst_idxs = [(2, 3, 9, 4), (2, 3, 9, 4), (0, 1, 4, 2), (3, 0, 2, 6)]
+    dst_params = [(i, j, 2), (k, l, 1), (m, n, 3), (o, p, 4)]
+
+    test_kv = interpolate.align_torsion_idxs_and_params(src_idxs, src_params, dst_idxs, dst_params)
+
+    ref_kv = {
+        ((2, 3, 9, 4), (a, b, 2), (i, j, 2)),
+        ((2, 1, 4, 3), (c, d, 1), (0, d, 1)),
+        ((0, 1, 4, 2), (e, f, 3), (m, n, 3)),
+        ((0, 1, 4, 2), (g, h, 1), (0, h, 1)),
+        ((2, 3, 9, 4), (0, l, 1), (k, l, 1)),
+        ((3, 0, 2, 6), (0, p, 4), (o, p, 4)),
+    }
+
+    assert test_kv == ref_kv
+
+
+def test_align_nonbonded():
+    """
+    Test that we can align idxs and parameters for nonbonded forces. For decoupled
+    nonbonded interactions, we should set q_ij, s_ij, and e_ij all to zero.
+    """
+    a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r = np.random.rand(18)
+
+    src_idxs = [(0, 3), (0, 2), (2, 3), (2, 4)]
+    src_params = [(a, b, c), (d, e, f), (g, h, i), (j, k, l)]
+    dst_idxs = [(0, 2), (4, 5)]
+    dst_params = [(m, n, o), (p, q, r)]
+
+    test_kv = interpolate.align_nonbonded_idxs_and_params(src_idxs, src_params, dst_idxs, dst_params)
+
+    ref_kv = {
+        ((0, 3), (a, b, c), (0, 0, 0)),
+        ((0, 2), (d, e, f), (m, n, o)),
+        ((2, 3), (g, h, i), (0, 0, 0)),
+        ((2, 4), (j, k, l), (0, 0, 0)),
+        ((4, 5), (0, 0, 0), (p, q, r)),
+    }
+
+    assert test_kv == ref_kv
+
+
+def test_align_chiral_atoms():
+    """
+    Test that we can align idxs and parameters for chiral atom restraints. We should
+    expect that force constants are set to zero.
+    """
+    a, b, c, d, e = np.random.rand(5)
+
+    src_idxs = [(0, 3, 4, 5), (0, 4, 3, 5), (4, 3, 5, 6)]
+    src_params = [a, b, c]
+    dst_idxs = [(0, 4, 3, 5), (4, 3, 5, 7)]
+    dst_params = [d, e]
+
+    test_kv = interpolate.align_chiral_atom_idxs_and_params(src_idxs, src_params, dst_idxs, dst_params)
+
+    ref_kv = {
+        ((0, 3, 4, 5), a, 0),
+        ((0, 4, 3, 5), b, d),
+        ((4, 3, 5, 6), c, 0),
+        ((4, 3, 5, 7), 0, e),
+    }
+
+    assert test_kv == ref_kv
+
+
+def test_align_chiral_bonds():
+    """
+    Similar to the chiral_atoms test, except that we check to see if deduplication
+    is using the sign information correctly.
```

Comment:
nit: might be explicit what "correctly" means here (e.g. as part of the alignment key)